From 73e81c42b52b1f4b84dd7e249c8e6824c3c8d7c6 Mon Sep 17 00:00:00 2001
From: ctang <ctang.ma@gmail.com>
Date: Tue, 27 Oct 2015 08:16:24 -0400
Subject: [PATCH 0714/1164] CDH-42679 : HIVE-12259: Command containing
 semicolon is broken in Beeline (Chaoyu Tang,
 reviewed by Ferdinand Xu)

Change-Id: Ic9a43120662166d18a4fbb8ced35607ebc696ce8
---
 .../src/java/org/apache/hive/beeline/BeeLine.java  |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/beeline/src/java/org/apache/hive/beeline/BeeLine.java b/beeline/src/java/org/apache/hive/beeline/BeeLine.java
index 12e1e46..a710cf2 100644
--- a/beeline/src/java/org/apache/hive/beeline/BeeLine.java
+++ b/beeline/src/java/org/apache/hive/beeline/BeeLine.java
@@ -1154,8 +1154,8 @@ boolean dispatch(String line) {
     }
 
     if (isBeeLine) {
-      if (line.startsWith(COMMAND_PREFIX) && !line.contains(";")) {
-        // handle the case "!cmd" for beeline
+      if (line.startsWith(COMMAND_PREFIX)) {
+        // handle SQLLine command in beeline which starts with ! and does not end with ;
         return execCommandWithPrefix(line);
       } else {
         return commands.sql(line, getOpts().getEntireLineAsCommand());
-- 
1.7.9.5

